pkg/security/probe/syscall_stats.go

```diff
@@ -124,10 +170,29 @@ func (sm *SyscallMonitor) CollectStats(collector SyscallStatsCollector) error {
 			}
 		}
 
-		if err := collector.Count(processSyscall.Process, Syscall(processSyscall.ID), value); err != nil {
+		if err := collector.CountSyscall(processSyscall.Process, Syscall(processSyscall.ID), value); err != nil {
+			return err
+		}
+	}
+	if mapIterator.Err() != nil {
+		log.Debugf("couldn't iterate over %s: %v", buffer.String(), mapIterator.Err())
+	}
+
+	mapIterator = execBuffer.Iterate()
+	for mapIterator.Next(&processPath, &value) {
+		if !processPath.IsEmpty() {
+			if err := execBuffer.Delete(&processPath.PathRaw); err != nil {
+				log.Debug(err)
+			}
+		}
+
+		if err := collector.CountExec(processPath.Path, value); err != nil {
```

Comment:
should we move this within `if !processPatth.IsEmpty()`

Reply:
👍